src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.lib.DataProcessingException;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String URL = "jdbc:mysql://localhost:3306/your_schema?serverTimezone=UTC";
+    private static final String USER = "your_user";
+    private static final String PASSWORD = "your_password";
+
+    @Override
+    public Book create(Book book) {
+        String insertQuery = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement statement = connection.prepareStatement(insertQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String selectQuery = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement statement = connection.prepareStatement(selectQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(mapResultSetToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String selectAllQuery = "SELECT * FROM books";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                Statement statement = connection.createStatement();

```

Comment:
Instead of using `Statement`, use `PreparedStatement` for consistency and potentially better performance. Also, consider extracting the query into a separate variable outside of the try block.